src/platform/STM32/timer_hal.c

```diff
@@ -1251,4 +1251,25 @@ HAL_StatusTypeDef DMA_SetCurrDataCounter(TIM_HandleTypeDef *htim, uint32_t Chann
     /* Return function status */
     return HAL_OK;
 }
+
+void timerDeInitTimer(TIM_TypeDef *timer)
+{
+    LL_TIM_DeInit(timer);
+}
+
+void timerSetTimerPeriod(TIM_TypeDef *timer, uint32_t period)
```

Comment:
(technically, this should use __HAL_TIM_SET_AUTORELOAD. But timer handle would be necessary and getting it is quite expensive.
Whole timer HAL implementation is suboptimal - it should be based around  TIM_HandleTypeDef and not TIM_TypeDef  )

Reply:
If you go a step further and look into the LL_TIM code - it mostly does handle->instance->X where instance is TIM_TypeDef so is essentially just glorified STD PERIPH.